opm/simulators/wells/WellHelpers.cpp

```diff
@@ -133,9 +133,9 @@ Scalar computeHydrostaticCorrection(const Scalar well_ref_depth, const Scalar vf
     return dp;
 }
 
-template<typename Scalar, typename Comm>
-void sumDistributedWellEntries(Dune::DynamicMatrix<Scalar>& mat,
-                               Dune::DynamicVector<Scalar>& vec,
+template<typename MatrixType, typename VectorType, typename Comm>
+void sumDistributedWellEntries(MatrixType& mat,
+                               VectorType& vec,
                                const Comm& comm)
 {
     // DynamicMatrix does not use one contiguous array for storing the data
```

Comment:
I think for StandardWell it is still DyanmicMatrix.

Reply:
> Nice! Now I am not sure why I did it this way in the first place. Good that you looked at it.
> 
> So this is now only called with FieldMatrix? That is known to have contiguous storage. Hence we might be able to optimize the method further. Maybe this will do the same
> 
> ```
> comm.sum(&(*mat.begin().begin()), m.mat_cols*m.mat_rows);
> comm.sum(&(*vec.begin()), vec.size);
> ```
> 
> We might want to check which version is faster. I also did not know `mat_cols` before 🙈

I tried that, it does not work - so I suggest to leave this method as it is.

Reply:
I tried your approach and also
```
comm.sum(mat.mat_access(), mat.mat_cols()*mat.mat_rows());
comm.sum(vec.data(), vec.size());
```
but both did not work, so I suggest to leave this as it is.